include/service/entities/plugin_service.js

```diff
@@ -1035,45 +1042,74 @@ module.exports = function PluginServiceModule(pb) {
 
         pb.log.debug("PluginService:[INIT] Beginning initialization of %s (%s)", plugin.name, plugin.uid);
 
+        var tasks = this.getInitTasksForPlugin(plugin);
+
+        async.series(tasks, function(err, results) {
+            //cleanup on error
+            if (util.isError(err) && details && details.uid) {
+                delete ACTIVE_PLUGINS[details.uid];
```

Comment:
The ```details``` variable is not defined anywhere.  You would also not need the check for the uid in the ```if``` statement because if you make it to here then you have a plugin UID available on the ```plugin``` variable.